regression/cbmc/Overflow_Leftshift1/main.c

```diff
@@ -23,5 +25,9 @@ int main()
   // not an overflow in ANSI-C, but undefined in C99
   s = 1 << (sizeof(int) * 8 - 1);
 
+  // overflow in an expression where operand and distance types are different
+  uint32_t u32;
+  int64_t i64 = (int64_t)u32 << 32;
```

Comment:
This is clever.  For those of us who struggle to remember the relative precedence of cast and shift and the C type promotion rules, would it be possible to be write this as:

```
((int64_t)u32) << 32
```

Reply:
Ah, hold on, is that the point, this is a C89 / C99 difference?

Reply:
Done - no, this C89/C99 difference was not mean to be important, it's just relevant that the types of the operand to be shifted and the type of the shift distance need to differ.

Reply:
Thanks!